frame/ethereum/src/lib.rs

```diff
@@ -225,6 +238,54 @@ impl<T: Trait> Module<T> {
 	pub fn transaction_status(hash: H256) -> Option<TransactionStatus> {
 		TransactionStatuses::get(hash)
 	}
+	
+	pub fn transaction_by_hash(hash: H256) -> Option<FullTransaction> {
+		let (block_hash, transaction_index) = match Transactions::get(hash) {
+			Some(a) => a,
+			_ => return None,
+		};
+
+		let transaction_status = match TransactionStatuses::get(hash) {
+			Some(a) => a,
+			_ => return None,
+		};
+
+		let (block,_receipt) = match BlocksAndReceipts::get(block_hash) {
+			Some(a) => a,
+			_ => return None,
+		};
+
+		let transaction = &block.transactions[transaction_index as usize];
+
+		Some(FullTransaction {
```

Comment:
Do you think it is worth it if we set the runtime API as fetching transaction, receipt and status separately, and then construct the `FullTransaction` only in RPC?

Reply:
Agreed, it will look much cleaner in the Runtime side.